pkgs/development/libraries/libptytty/default.nix

```diff
@@ -4,6 +4,8 @@
 , cmake
 }:
 
+let isCross = stdenv.hostPlatform.system != stdenv.buildPlatform.system; in
```

Comment:
`stdenv.hostPlatform != stdenv.buildPlatform` is more commonly used.

Reply:
Both are used. Comparing the `attrs` or the system `string`. In some cases with meaningful difference, like this:
https://github.com/NixOS/nixpkgs/pull/202856/files#diff-7c03bad85e7df617833680bacd05732b811a11a583def009880b14a3edf8dd6dR144

Attrs content is:

> nix-repl> stdenv.hostPlatform
> { aesSupport = false; avx2Support = false; avx512Support = false; avxSupport = false; canExecute = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:35:20»; config = "x86_64-unknown-linux-gnu"; darwinArch = "x86_64"; darwinMinVersion = "10.12"; darwinMinVersionVariable = null; darwinPlatform = null; darwinSdkVersion = "10.12"; efiArch = "x64"; emulator = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:206:20»; emulatorAvailable = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:204:29»; extensions = { ... }; fma4Support = false; fmaSupport = false; gcc = { ... }; is32bit = false; is64bit = true; isAarch = false; isAarch32 = false; isAarch64 = false; isAbiElfv2 = false; isAlpha = false; isAndroid = false; isArmv7 = false; isAvr = false; isBSD = false; isBigEndian = false; isCompatible = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:39:22»; isCygwin = false; isDarwin = false; isEfi = true; isFreeBSD = false; isGenode = false; isGhcjs = false; isGnu = true; isILP32 = false; isJavaScript = false; isLinux = true; isLittleEndian = true; isM68k = false; isMacOS = false; isMicroBlaze = false; isMinGW = false; isMips = false; isMips32 = false; isMips64 = false; isMips64n32 = false; isMips64n64 = false; isMmix = false; isMsp430 = false; isMusl = false; isNetBSD = false; isNone = false; isOpenBSD = false; isOr1k = false; isPower = false; isPower64 = false; isRedox = false; isRiscV = false; isRiscV32 = false; isRiscV64 = false; isRx = false; isS390 = false; isS390x = false; isSparc = false; isStatic = false; isSunOS = false; isUClibc = false; isUnix = true; isVc4 = false; isWasi = false; isWasm = false; isWindows = false; isi686 = false; isiOS = false; isx86 = true; isx86_32 = false; isx86_64 = true; libc = "glibc"; linker = "bfd"; linux-kernel = { ... }; linuxArch = "x86_64"; parsed = { ... }; qemuArch = "x86_64"; rustc = { ... }; sse3Support = false; sse4_1Support = false; sse4_2Support = false; sse4_aSupport = false; ssse3Support = false; system = "x86_64-linux"; uname = { ... }; useAndroidPrebuilt = false; useiOSPrebuilt = false; }
> 
> nix-repl> stdenv.buildPlatform
> { aesSupport = false; avx2Support = false; avx512Support = false; avxSupport = false; canExecute = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:35:20»; config = "x86_64-unknown-linux-gnu"; darwinArch = "x86_64"; darwinMinVersion = "10.12"; darwinMinVersionVariable = null; darwinPlatform = null; darwinSdkVersion = "10.12"; efiArch = "x64"; emulator = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:206:20»; emulatorAvailable = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:204:29»; extensions = { ... }; fma4Support = false; fmaSupport = false; gcc = { ... }; is32bit = false; is64bit = true; isAarch = false; isAarch32 = false; isAarch64 = false; isAbiElfv2 = false; isAlpha = false; isAndroid = false; isArmv7 = false; isAvr = false; isBSD = false; isBigEndian = false; isCompatible = «lambda @ /nix/store/gm8c760jh9w22vybdj99g1ciz2npmj7c-source/lib/systems/default.nix:39:22»; isCygwin = false; isDarwin = false; isEfi = true; isFreeBSD = false; isGenode = false; isGhcjs = false; isGnu = true; isILP32 = false; isJavaScript = false; isLinux = true; isLittleEndian = true; isM68k = false; isMacOS = false; isMicroBlaze = false; isMinGW = false; isMips = false; isMips32 = false; isMips64 = false; isMips64n32 = false; isMips64n64 = false; isMmix = false; isMsp430 = false; isMusl = false; isNetBSD = false; isNone = false; isOpenBSD = false; isOr1k = false; isPower = false; isPower64 = false; isRedox = false; isRiscV = false; isRiscV32 = false; isRiscV64 = false; isRx = false; isS390 = false; isS390x = false; isSparc = false; isStatic = false; isSunOS = false; isUClibc = false; isUnix = true; isVc4 = false; isWasi = false; isWasm = false; isWindows = false; isi686 = false; isiOS = false; isx86 = true; isx86_32 = false; isx86_64 = true; libc = "glibc"; linker = "bfd"; linux-kernel = { ... }; linuxArch = "x86_64"; parsed = { ... }; qemuArch = "x86_64"; rustc = { ... }; sse3Support = false; sse4_1Support = false; sse4_2Support = false; sse4_aSupport = false; ssse3Support = false; system = "x86_64-linux"; uname = { ... }; useAndroidPrebuilt = false; useiOSPrebuilt = false; }

I prefer the system approach because I can think about it better (without having to consider all the possibilities/options in attrs). And it just works. But in case, I'm being naive, just let me know and I'll change it.

Reply:
According to https://github.com/yusiwen/libptytty/issues/2. TTY_GID_SUPPORT is tested by TRY_RUN, maybe `canExecute` is a more proper condition.

Reply:
I've made the change to `canExecute`. But the sad part is building/testing it. Building staging is not fun. (It is still building.)

Reply:
> > I actually think we need some sort of `stdenv.isCross` function. Because this isCross is always popping up in nixpkgs.
> 
> #44367

Thanks for the reference.
I don't like his reasoning though.
I still think `isCross` would be helpful.
`ripgrep` for `stdenv.buildPlatform != stdenv.hostPlatform` and the list is huge!
Also there is uncertainty on which way is correct to check for `isCross`. (Which just happened here.)